src/main/java/wooteco/subway/domain/Sections.java

```diff
@@ -0,0 +1,218 @@
+        cutSection(includingSection, newSection, anotherNewSection);
+    }
+
+    private void cutSection(Section includingSection, Section newSection, Section anotherNewSection) {
+        values.remove(includingSection);
+        values.add(newSection);
+        values.add(anotherNewSection);
+    }
+
+    public void removeSectionByStation(Station station) {
+        validateContainsTargetStation(station);
+        if (isEndSection(station)) {
+            removeEndSection(station);
+            return;
+        }
+        removeMiddleSection(station);
+    }
+
+    private boolean isEndSection(Station station) {
+        List<Station> sortedStations = getSortedStations();
+        int index = sortedStations.indexOf(station);
+        return index == 0 || index == sortedStations.size() - 1;
+    }
+
+    private void validateContainsTargetStation(Station station) {
+        if (values.stream()
+                .noneMatch(section -> section.containStation(station))) {
+            throw new SectionNotFoundException("요청하는 역을 포함하는 구간이 없습니다.");
+        }
+    }
+
+    private void removeEndSection(Station station) {
+        Section targetSection = values.stream()
+                .filter(savedSection -> savedSection.containStation(station))
+                .findFirst()
+                .orElseThrow(() -> new SectionNotFoundException("요청하는 역을 포함하는 구간이 없습니다."));
+        values.remove(targetSection);
+    }
+
+    private void removeMiddleSection(Station station) {
+        Section upSection = findUpSection(station);
+        Section downSection = findDownSection(station);
+        Section newSection = new Section(upSection.getUpStation(), downSection.getDownStation(),
+                upSection.getDistance() + downSection.getDistance());
+        values.remove(upSection);
+        values.remove(downSection);
+        values.add(newSection);
+    }
+
+    private Section findUpSection(Station station) {
+        return values.stream()
+                .filter(savedSection -> savedSection.getDownStation().hasSameName(station))
+                .findAny()
+                .orElseThrow(() -> new SectionNotFoundException("요청하는 역을 포함하는 구간이 없습니다."));
+    }
+
+    private Section findDownSection(Station station) {
+        return values.stream()
+                .filter(savedSection -> savedSection.getUpStation().hasSameName(station))
+                .findAny()
+                .orElseThrow(() -> new SectionNotFoundException("요청하는 역을 포함하는 구간이 없습니다."));
+    }
+
+    public List<Section> getNotContainSections(Sections sections) {
+        return values.stream()
+                .filter(sections::doesNotContain)
+                .collect(Collectors.toUnmodifiableList());
+    }
+
+    private boolean doesNotContain(Section target) {
+        return values.stream()
+                .noneMatch(section -> section.containStation(target.getUpStation()) && section.containStation(
+                        target.getDownStation()));
```

Comment:
```suggestion
        return values.stream()
                .noneMatch(section -> section.containStation(target.getUpStation())
                        && section.containStation(target.getDownStation()));
```
이렇게 변경하는건 어떠신가요? predicate 부분이 조금 더 명확하게 드러나도록 해보았어요.
개인적인 생각이니 오찌가 고민해주세요!ㅎㅎ